File: src/lessons/bat/string1/AltPairsEntity.java
package lessons.bat.string1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class AltPairsEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(altPairs((String)param[0])));
    }
  }

  /* BEGIN TEMPLATE */
  public String altPairs(String str)
  {
    /* BEGIN SOLUTION */
    String res = "";
    for (int i = 0; i < str.length(); i += 4)
      res += str.substring(i, Math.min(i + 2, str.length()));
    return res;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/bat/string1/AltPairsEntity.scala
package lessons.bat.string1

import plm.core.ValueSerializer
import plm.universe.bat.BatEntity

class AltPairsEntity extends BatEntity {

    override def run() {
      import ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
        setTestResult(i, serialize(altPairs(param(0).asInstanceOf[String])))
      }
    }

    /* BEGIN TEMPLATE */
    def altPairs(str: String): String = {
        /* BEGIN SOLUTION */
        var res = ""
        for (i <- 0 to (str.length - 1) by 4)
            res ++= str.substring(i, Math.min(i + 2, str.length))
        return res
        /* END SOLUTION */
    }
    /* END TEMPLATE */
}
